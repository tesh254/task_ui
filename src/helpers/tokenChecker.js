import jwt_decode from "jwt-decode";

export default () => {
  if (localStorage.getItem("jwt_token")) {
    try {
      const decoded = jwt_decode(localStorage.getItem("jwt_token"));

      const currentTime = Date.now() / 1000;

      if (decoded.exp < currentTime) {
        localStorage.clear();
      }
    } catch (e) {
      localStorage.clear();
    }
  } else {
    localStorage.clear();
  }
};
