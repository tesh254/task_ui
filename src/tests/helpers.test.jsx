import setAuthToken from "../helpers/setAuthToken";
import tokenChecker from "../helpers/tokenChecker";

describe("test helpers", () => {
  it("should test set auth token function", () => {
    const result = setAuthToken(false);
    expect(result).toMatchSnapshot();
  });

  it("should test set auth token with token present", () => {
    const result = setAuthToken("TOkne");

    expect(result).toMatchSnapshot();
  });

  it("test tokenChecker method with token in localstorage", () => {
    const result = tokenChecker();

    expect(result).toMatchSnapshot();
  });
});
