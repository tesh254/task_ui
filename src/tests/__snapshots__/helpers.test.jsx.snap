// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test helpers should test set auth token function 1`] = `undefined`;

exports[`test helpers should test set auth token with token present 1`] = `undefined`;

exports[`test helpers test tokenChecker method with token in localstorage 1`] = `undefined`;
